tcs-persistence/src/test/java/com/transformuk/hee/tis/tcs/service/repository/PostRepositoryTest.java

```diff
@@ -172,6 +173,72 @@ public void findPostsAndPlacementsByProgrammeIdAndSpecialtyIdShouldFindPostsLink
               .getForenames()));
 
     }
+  }
+
+  @Test
+  public void testClearOldPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
+    entityManager.persist(post2);
+    entityManager.flush();
+
+    postRepository.clearOldPostReferences(post1.getId());
+    entityManager.flush();
+    entityManager.clear();
+
+    Post updatedPost = entityManager.find(Post.class, post2.getId());
+    Assert.assertNull(updatedPost.getOldPost());
+  }
+
+  @Test
+  public void testClearNewPostReferencesInPostTableWhenDelete() {
+    Post post1 = new Post();
+    entityManager.persist(post1);
+
+    Post post2 = new Post();
+    post2.setOldPost(post1);
```

Comment:
I think this line should be `post2.setNewPost(post1);`

Reply:
Changed